### ISSUE TITLE
在3090上训练达不到论文中的准确度

### ISSUE THREAD

**Issue by luxilandu** [CNJianLiu/Diff9D#20]
作者，你好，直接使用你仓库中提供的train.py训练模型，应该对应的是论文中的这个部分吧

<img width="1586" height="304" alt="Image" src="https://github.com/user-attachments/assets/e4b55d14-917b-4b59-82c7-03cc00568d22" />

我在3090上训练，几乎每20轮都进行测试，但是我没有得到论文中的结果，不知道问题在哪

下面这个链接包含了训练日志以及REAL275的测试结果

https://drive.google.com/file/d/1fZk6BNA1nJhzSQmjve5CYyb-27O0NrCZ/view?usp=drive_link

**Comment by luxilandu** [comment]
I suspect that the dataset may not have been fully downloaded. Could you please confirm whether the number of images and object models used for training is consistent with that shown in the figures?

<img width="2505" height="341" alt="Image" src="https://github.com/user-attachments/assets/19540cab-1518-4b4b-a5ff-b4bc42f3c0af" />

**Comment by DrLazywh** [comment]
兄弟我在A800上也跑不到，此外数据集的情况我也和你一样。我觉得数据集加载照片和模型数量和你一致，我觉得数据集应该没啥问题，应该是下载的完整的。
A800的结果：
<img width="483" height="213" alt="Image" src="https://github.com/user-attachments/assets/77888443-4e7a-4c57-945b-a6fedee35c8b" />

**Comment by CNJianLiu** [comment]
Hi, did you complete the total 1000 epochs? It seems like the intermediate experimental results in my memory. Also, I cannot open the above link.

**Comment by luxilandu** [comment]
> Hi, did you complete the total 1000 epochs? It seems like the intermediate experimental results in my memory. Also, I cannot open the above link.

I am writing to confirm the number of images and object models loaded during model training with the figure attached above.
I have finished training for 1000 epochs and run evaluations on nearly every checkpoint. The test results as well as full training logs across all 1000 epochs are available via the link provided.



<img width="2026" height="1215" alt="Image" src="https://github.com/user-attachments/assets/ac0844ab-af6b-4055-a198-ec8077594afc" />

**Comment by DrLazywh** [comment]
兄弟，你把权限开一下看看啊，设置了访问权限，申请了，需要你同意，或者你直接改成随便访问的

**Comment by luxilandu** [comment]
> 兄弟，你把权限开一下看看啊，设置了访问权限，申请了，需要你同意，或者你直接改成随便访问的

设置好了，任何人都可以访问

**Comment by luxilandu** [comment]
Dear author:
Why the result in the paper is different from the result in your test log? The answer is extremely important for me , because I have made a mind based on your architecture.

---+++   your log:

<img width="1262" height="762" alt="Image" src="https://github.com/user-attachments/assets/aa62190a-fde4-4eb3-8d3e-99e08b29eea9" />

---+++ your paper:

<img width="1175" height="304" alt="Image" src="https://github.com/user-attachments/assets/fd4b9fec-3e54-4d09-827e-12c347fca654" />

**Comment by Ersa-user** [comment]
作者您好，打扰了。我也在复现这个 issue 中提到的 Diff9D REAL275 指标问题。

我也在使用 release code 复现 Diff9D 在 REAL275 上的论文结果，目前主要问题是 3D IoU@75 明显低于论文表格中的结果，所以想请教一下是否需要修改训练配置或环境。

我当前使用的是 clean release code，从 epoch 1 开始重新训练，没有使用 resume。

当前数据加载情况：

CAMERA train images: 249127
models loaded: 1083

当前主要训练配置如下：

optimizer:
  type: Adam
  lr: 0.01
  betas: [0.5, 0.999]
  eps: 1e-6
  weight_decay: 0

lr_scheduler:
  type: StepLR
  step_size: 1
  gamma: 0.5

max_epoch: 1000
num_mini_batch_per_epoch: 2000

train_dataset:
  img_size: 192
  sample_num: 1024
  shift_range: 0.01
  use_shape_aug: true
  aug_bb_pro: 0.3
  aug_rt_pro: 0.3
  aug_bc_pro: 0.0
  aug_pc_pro: 0.0
  aug_pc_r: 0.002
  aug_nl_pro: 0.0

train_dataloader:
  syn_bs: 48
  num_workers: 48
  shuffle: true
  drop_last: true
  pin_memory: false
  use_fill_miss: true
  use_composed_img: false

test:
  img_size: 192
  sample_num: 1024

rd_seed: 1
checkpoint_epoch: -1

我的环境大致如下：

GPU: NVIDIA RTX 5090
Python: 3.10
nvidia-smi CUDA version: 13.0
Driver Version: 580.126.09

请问复现论文中 Diff9D 的 REAL275 结果时，是否需要修改上述配置？

我主要想确认：

1. 论文表格中的 Diff9D 结果是否就是使用 release code 默认配置训练得到的？
2. syn_bs=48、num_mini_batch_per_epoch=2000、lr=0.01、StepLR(gamma=0.5) 是否需要根据 GPU 类型或 batch size 调整？
3. 如果中途 resume，是否需要同时恢复 optimizer 和 lr_scheduler，还是建议必须从 epoch 1 连续训练到 epoch 1000？
4. GPU 型号、Python 版本、CUDA / PyTorch 版本是否会明显影响 REAL275 的 3D IoU@75？
5. 是否有推荐的复现环境版本，例如 Python、PyTorch、CUDA、GPU 类型？
6. A800 上复现的 3D IoU@75 约为 41.0，这是否是 release code 下正常的复现水平，还是理论上应该接近论文中的 44.1？

谢谢作者。

**Comment by luxilandu** [comment]
> 作者您好，打扰了。我也在复现这个 issue 中提到的 Diff9D REAL275 指标问题。
> 
> 我也在使用 release code 复现 Diff9D 在 REAL275 上的论文结果，目前主要问题是 3D IoU@75 明显低于论文表格中的结果，所以想请教一下是否需要修改训练配置或环境。
> 
> 我当前使用的是 clean release code，从 epoch 1 开始重新训练，没有使用 resume。
> 
> 当前数据加载情况：
> 
> CAMERA train images: 249127 models loaded: 1083
> 
> 当前主要训练配置如下：
> 
> optimizer: type: Adam lr: 0.01 betas: [0.5, 0.999] eps: 1e-6 weight_decay: 0
> 
> lr_scheduler: type: StepLR step_size: 1 gamma: 0.5
> 
> max_epoch: 1000 num_mini_batch_per_epoch: 2000
> 
> train_dataset: img_size: 192 sample_num: 1024 shift_range: 0.01 use_shape_aug: true aug_bb_pro: 0.3 aug_rt_pro: 0.3 aug_bc_pro: 0.0 aug_pc_pro: 0.0 aug_pc_r: 0.002 aug_nl_pro: 0.0
> 
> train_dataloader: syn_bs: 48 num_workers: 48 shuffle: true drop_last: true pin_memory: false use_fill_miss: true use_composed_img: false
> 
> test: img_size: 192 sample_num: 1024
> 
> rd_seed: 1 checkpoint_epoch: -1
> 
> 我的环境大致如下：
> 
> GPU: NVIDIA RTX 5090 Python: 3.10 nvidia-smi CUDA version: 13.0 Driver Version: 580.126.09
> 
> 请问复现论文中 Diff9D 的 REAL275 结果时，是否需要修改上述配置？
> 
> 我主要想确认：
> 
> 1. 论文表格中的 Diff9D 结果是否就是使用 release code 默认配置训练得到的？
> 2. syn_bs=48、num_mini_batch_per_epoch=2000、lr=0.01、StepLR(gamma=0.5) 是否需要根据 GPU 类型或 batch size 调整？
> 3. 如果中途 resume，是否需要同时恢复 optimizer 和 lr_scheduler，还是建议必须从 epoch 1 连续训练到 epoch 1000？
> 4. GPU 型号、Python 版本、CUDA / PyTorch 版本是否会明显影响 REAL275 的 3D IoU@75？
> 5. 是否有推荐的复现环境版本，例如 Python、PyTorch、CUDA、GPU 类型？
> 6. A800 上复现的 3D IoU@75 约为 41.0，这是否是 release code 下正常的复现水平，还是理论上应该接近论文中的 44.1？
> 
> 谢谢作者。

There is an interesting pause in the training process at epoch51.

**Comment by Ersa-user** [comment]
> > 作者您好，打扰了。我也在复现这个 issue 中提到的 Diff9D REAL275 指标问题。
> > 我也在使用 release code 复现 Diff9D 在 REAL275 上的论文结果，目前主要问题是 3D IoU@75 明显低于论文表格中的结果，所以想请教一下是否需要修改训练配置或环境。
> > 我当前使用的是 clean release code，从 epoch 1 开始重新训练，没有使用 resume。
> > 当前数据加载情况：
> > CAMERA train images: 249127 models loaded: 1083
> > 当前主要训练配置如下：
> > optimizer: type: Adam lr: 0.01 betas: [0.5, 0.999] eps: 1e-6 weight_decay: 0
> > lr_scheduler: type: StepLR step_size: 1 gamma: 0.5
> > max_epoch: 1000 num_mini_batch_per_epoch: 2000
> > train_dataset: img_size: 192 sample_num: 1024 shift_range: 0.01 use_shape_aug: true aug_bb_pro: 0.3 aug_rt_pro: 0.3 aug_bc_pro: 0.0 aug_pc_pro: 0.0 aug_pc_r: 0.002 aug_nl_pro: 0.0
> > train_dataloader: syn_bs: 48 num_workers: 48 shuffle: true drop_last: true pin_memory: false use_fill_miss: true use_composed_img: false
> > test: img_size: 192 sample_num: 1024
> > rd_seed: 1 checkpoint_epoch: -1
> > 我的环境大致如下：
> > GPU: NVIDIA RTX 5090 Python: 3.10 nvidia-smi CUDA version: 13.0 Driver Version: 580.126.09
> > 请问复现论文中 Diff9D 的 REAL275 结果时，是否需要修改上述配置？
> > 我主要想确认：
> > 
> > 1. 论文表格中的 Diff9D 结果是否就是使用 release code 默认配置训练得到的？
> > 2. syn_bs=48、num_mini_batch_per_epoch=2000、lr=0.01、StepLR(gamma=0.5) 是否需要根据 GPU 类型或 batch size 调整？
> > 3. 如果中途 resume，是否需要同时恢复 optimizer 和 lr_scheduler，还是建议必须从 epoch 1 连续训练到 epoch 1000？
> > 4. GPU 型号、Python 版本、CUDA / PyTorch 版本是否会明显影响 REAL275 的 3D IoU@75？
> > 5. 是否有推荐的复现环境版本，例如 Python、PyTorch、CUDA、GPU 类型？
> > 6. A800 上复现的 3D IoU@75 约为 41.0，这是否是 release code 下正常的复现水平，还是理论上应该接近论文中的 44.1？
> > 
> > 谢谢作者。
> 
> There is an interesting pause in the training process at epoch51.
谢谢作者提醒。我先重新从 epoch 1 连续训练一次。

上一次训练过程中确实有中断，所以之前的日志可能不能代表严格连续训练的结果。  
我会先完成一次不使用 resume 的完整训练，再回到这个 issue 更新结果。

谢谢。

**Comment by luxilandu** [comment]
> > > 作者您好，打扰了。我也在复现这个 issue 中提到的 Diff9D REAL275 指标问题。
> > > 我也在使用 release code 复现 Diff9D 在 REAL275 上的论文结果，目前主要问题是 3D IoU@75 明显低于论文表格中的结果，所以想请教一下是否需要修改训练配置或环境。
> > > 我当前使用的是 clean release code，从 epoch 1 开始重新训练，没有使用 resume。
> > > 当前数据加载情况：
> > > CAMERA train images: 249127 models loaded: 1083
> > > 当前主要训练配置如下：
> > > optimizer: type: Adam lr: 0.01 betas: [0.5, 0.999] eps: 1e-6 weight_decay: 0
> > > lr_scheduler: type: StepLR step_size: 1 gamma: 0.5
> > > max_epoch: 1000 num_mini_batch_per_epoch: 2000
> > > train_dataset: img_size: 192 sample_num: 1024 shift_range: 0.01 use_shape_aug: true aug_bb_pro: 0.3 aug_rt_pro: 0.3 aug_bc_pro: 0.0 aug_pc_pro: 0.0 aug_pc_r: 0.002 aug_nl_pro: 0.0
> > > train_dataloader: syn_bs: 48 num_workers: 48 shuffle: true drop_last: true pin_memory: false use_fill_miss: true use_composed_img: false
> > > test: img_size: 192 sample_num: 1024
> > > rd_seed: 1 checkpoint_epoch: -1
> > > 我的环境大致如下：
> > > GPU: NVIDIA RTX 5090 Python: 3.10 nvidia-smi CUDA version: 13.0 Driver Version: 580.126.09
> > > 请问复现论文中 Diff9D 的 REAL275 结果时，是否需要修改上述配置？
> > > 我主要想确认：
> > > 
> > > 1. 论文表格中的 Diff9D 结果是否就是使用 release code 默认配置训练得到的？
> > > 2. syn_bs=48、num_mini_batch_per_epoch=2000、lr=0.01、StepLR(gamma=0.5) 是否需要根据 GPU 类型或 batch size 调整？
> > > 3. 如果中途 resume，是否需要同时恢复 optimizer 和 lr_scheduler，还是建议必须从 epoch 1 连续训练到 epoch 1000？
> > > 4. GPU 型号、Python 版本、CUDA / PyTorch 版本是否会明显影响 REAL275 的 3D IoU@75？
> > > 5. 是否有推荐的复现环境版本，例如 Python、PyTorch、CUDA、GPU 类型？
> > > 6. A800 上复现的 3D IoU@75 约为 41.0，这是否是 release code 下正常的复现水平，还是理论上应该接近论文中的 44.1？
> > > 
> > > 谢谢作者。
> > 
> > 
> > There is an interesting pause in the training process at epoch51.
> > 谢谢作者提醒。我先重新从 epoch 1 连续训练一次。
> 
> 上一次训练过程中确实有中断，所以之前的日志可能不能代表严格连续训练的结果。 我会先完成一次不使用 resume 的完整训练，再回到这个 issue 更新结果。
> 
> 谢谢。

Hi, I am not the author. I meant that there is a pause in the training process at epoch51 in author's training log. By the way, have you reproduced other results reported in the paper?

**Comment by Ersa-user** [comment]
> > > > 作者您好，打扰了。我也在复现这个 issue 中提到的 Diff9D REAL275 指标问题。
> > > > 我也在使用 release code 复现 Diff9D 在 REAL275 上的论文结果，目前主要问题是 3D IoU@75 明显低于论文表格中的结果，所以想请教一下是否需要修改训练配置或环境。
> > > > 我当前使用的是 clean release code，从 epoch 1 开始重新训练，没有使用 resume。
> > > > 当前数据加载情况：
> > > > CAMERA train images: 249127 models loaded: 1083
> > > > 当前主要训练配置如下：
> > > > optimizer: type: Adam lr: 0.01 betas: [0.5, 0.999] eps: 1e-6 weight_decay: 0
> > > > lr_scheduler: type: StepLR step_size: 1 gamma: 0.5
> > > > max_epoch: 1000 num_mini_batch_per_epoch: 2000
> > > > train_dataset: img_size: 192 sample_num: 1024 shift_range: 0.01 use_shape_aug: true aug_bb_pro: 0.3 aug_rt_pro: 0.3 aug_bc_pro: 0.0 aug_pc_pro: 0.0 aug_pc_r: 0.002 aug_nl_pro: 0.0
> > > > train_dataloader: syn_bs: 48 num_workers: 48 shuffle: true drop_last: true pin_memory: false use_fill_miss: true use_composed_img: false
> > > > test: img_size: 192 sample_num: 1024
> > > > rd_seed: 1 checkpoint_epoch: -1
> > > > 我的环境大致如下：
> > > > GPU: NVIDIA RTX 5090 Python: 3.10 nvidia-smi CUDA version: 13.0 Driver Version: 580.126.09
> > > > 请问复现论文中 Diff9D 的 REAL275 结果时，是否需要修改上述配置？
> > > > 我主要想确认：
> > > > 
> > > > 1. 论文表格中的 Diff9D 结果是否就是使用 release code 默认配置训练得到的？
> > > > 2. syn_bs=48、num_mini_batch_per_epoch=2000、lr=0.01、StepLR(gamma=0.5) 是否需要根据 GPU 类型或 batch size 调整？
> > > > 3. 如果中途 resume，是否需要同时恢复 optimizer 和 lr_scheduler，还是建议必须从 epoch 1 连续训练到 epoch 1000？
> > > > 4. GPU 型号、Python 版本、CUDA / PyTorch 版本是否会明显影响 REAL275 的 3D IoU@75？
> > > > 5. 是否有推荐的复现环境版本，例如 Python、PyTorch、CUDA、GPU 类型？
> > > > 6. A800 上复现的 3D IoU@75 约为 41.0，这是否是 release code 下正常的复现水平，还是理论上应该接近论文中的 44.1？
> > > > 
> > > > 谢谢作者。
> > > 
> > > 
> > > There is an interesting pause in the training process at epoch51.
> > > 谢谢作者提醒。我先重新从 epoch 1 连续训练一次。
> > 
> > 
> > 上一次训练过程中确实有中断，所以之前的日志可能不能代表严格连续训练的结果。 我会先完成一次不使用 resume 的完整训练，再回到这个 issue 更新结果。
> > 谢谢。
> 
> Hi, I am not the author. I meant that there is a pause in the training process at epoch51 in author's training log. By the way, have you reproduced other results reported in the paper?

目前我主要在检查 REAL275 的复现情况，最大的差距仍然是 3D IoU@75，复现出来的其他结果相差不是很大。

我会先完成一次不使用 resume 的完整连续训练，再重新对比最终指标。
如果你或其他人已经复现了论文中的其他结果，也欢迎告诉我对应的设置和环境，感谢。
